cli/tests/integration/lsp_tests.rs

```diff
@@ -2818,12 +2923,8 @@ fn lsp_code_actions_deadlock() {
   assert!(maybe_err.is_none());
   assert!(maybe_res.is_some());
 
-  for _ in 0..3 {
-    let (method, _) = client.read_notification::<Value>().unwrap();
-    assert_eq!(method, "textDocument/publishDiagnostics");
-  }
+  read_diagnostics(&mut client);
 
-  assert!(client.queue_is_empty());
```

Comment:
I removed this check because it's now flaky because of the multiple `textDocument/didChange` notifications in a row above without waiting for diagnostics (it's actually now possible for lint or deps diagnostics to finish really quickly).